core/src/Kokkos_View.hpp

```diff
@@ -210,7 +210,7 @@ class View : public Impl::BasicViewFromTraits<DataType, Properties...>::type {
   using host_mirror_type =
       View<typename traits::non_const_data_type, typename traits::array_layout,
            Device<DefaultHostExecutionSpace,
-                  typename traits::host_mirror_space::memory_space>,
+                  typename traits::host_mirror_device::memory_space>,
```

Comment:
As I said for other classes this may need a deprecation guard.

Reply:
Don't you need to deprecate the `HostMirror` below?

Reply:
For the above, we likely do not need a deprecation guard since the behavior did not change: memory_space is still memory_space in either HostMirror implementation.